include/proto.h

```diff
@@ -494,11 +494,11 @@ enum {
 /* Item rarity designations used for different object presentations */
 typedef enum {
    ITEM_RARITY_GRADE_NORMAL       = 0,
-   ITEM_RARITY_GRADE_MAGIC        = 1,
+   ITEM_RARITY_GRADE_UNCOMMON     = 1,
    ITEM_RARITY_GRADE_RARE         = 2,
-   ITEM_RARITY_GRADE_LEGENDARY    = 3,
-   ITEM_RARITY_GRADE_UNIDENTIFIED = 4,
-   ITEM_RARITY_GRADE_CURSED       = 5,
+   ITEM_RARITY_GRADE_LEGENDARY    = 4,
+   ITEM_RARITY_GRADE_UNIDENTIFIED = 100,
+   ITEM_RARITY_GRADE_CURSED       = 200,
```

Comment:
I'm not sure why you changed these enum values.  They're basically opaque integers with no other meaning as far as I can tell.

Reply:
I can change them back but I figured it might be doing for 2 reasons
- after this PR, `viRarity` + count of attributes will determine the rarity of an item with attributes
  - Since enum legendary was set to 3 and the threshold for an item to be legendary was 4 I thought this might introduce confusion.  By making legendary always equal to 4 it just kind of standardizes it.  Not necessary.
 - if someone adds more grades or levels of rarity, the next number would be after cursed (5) but the current highest is legendary(4) so that would be (6).  
   - I felt like people might trip over this and perhaps to limit confusion we could jump to the next hundredth for categories since it is sending a full INT.

I don't know if this is a bad practice or not, I was just worried about confusing myself and locking myself into a structure that seemed logical (higher number = more rare) but then switched to unidentified, cursed, and then more rare.

Reply:
Changing values that are used in a network protocol is dangerous in general, because there can be existing clients out there that you break when you change the values.  It's not an issue in this particular case, since it hasn't launched yet, but it's something to keep in mind.  Let me know your preference.

Reply:
Understood.  I guess since we haven't done anything yet, we could do the current revision.  The only issue is if we add more grades in between rare and legendary or uncommon and rare it will fall apart again.  Maybe I should do by 10s? I get a IP addressing vibe from this when people design their networks to have space for expansion or "categories" of IP ranges (printers are 10-20, etc).

I guess as is works, or I can do every 10 so then 0 is normal, 10 is uncommon, 20 rare, etc.

From what I've done is brief 1:1 questioning with current players they are mostly interested in the unidentified rarity grade since that seems to be a huge pain point at the moment.  They really don't seem to have many questions about the higher rarity grades, at the moment anyway.

Reply:
Ugh, sorry for all the comments.

I think if I use bigger numbers the calculation for rarity will be messed up anyway.  As it stands is probably fine, and then if someone changes things they will have to adjust the count calculation anyway.

For example if rare is 20 then adding 3 item counts to it will not reach the next level.

In short, I don't think my idea will really scale well anyways.  We will need to modify the calculation function regardless.  So I guess the current state is fine as it's pseudo-intuitive?